log-replication/src/main/java/org/corfudb/logreplication/fsm/InLogEntrySyncState.java

```diff
@@ -0,0 +1,152 @@
+     * @param logEntrySender instance that will manage reading and sending log data between sites.
+     */
+    public InLogEntrySyncState(LogReplicationFSM logReplicationFSM, LogEntrySender logEntrySender) {
+        this.fsm = logReplicationFSM;
+        this.logEntrySender = logEntrySender;
+    }
+
+    @Override
+    public LogReplicationState processEvent(LogReplicationEvent event) throws IllegalTransitionException {
+        switch (event.getType()) {
+            case SNAPSHOT_SYNC_REQUEST:
+                cancelLogEntrySync("snapshot sync request.");
+                LogReplicationState snapshotSyncState = fsm.getStates().get(LogReplicationStateType.IN_SNAPSHOT_SYNC);
+                snapshotSyncState.setTransitionEventId(event.getEventID());
+                return snapshotSyncState;
+            case SYNC_CANCEL:
+                // If cancel was intended for current log entry sync task, cancel and transition to new state
+                // In the case of log entry sync, cancel is caused by an encountered trimmed exception or any
+                // other exception while reading/sending.
+                if (transitionEventId == event.getMetadata().getRequestId()) {
+                    cancelLogEntrySync("sync cancel.");
+                    LogReplicationState inSnapshotSyncState = fsm.getStates().get(LogReplicationStateType.IN_SNAPSHOT_SYNC);
+                    inSnapshotSyncState.setTransitionEventId(UUID.randomUUID());
+                    return inSnapshotSyncState;
+                }
+
+                log.warn("Log Entry Sync cancel for eventId {}, but running log entry sync for {}",
+                        event.getEventID(), transitionEventId);
+                return this;
+            case REPLICATION_STOP:
+                cancelLogEntrySync("replication being stopped.");
+                return fsm.getStates().get(LogReplicationStateType.INITIALIZED);
+            case REPLICATION_SHUTDOWN:
+                cancelLogEntrySync("replication terminated.");
+                return fsm.getStates().get(LogReplicationStateType.STOPPED);
+            case LOG_ENTRY_SYNC_CONTINUE:
+                // Snapshot sync is broken into multiple tasks, where each task sends a batch of messages
+                // corresponding to this snapshot sync. This is done to accommodate the case
+                // of multi-site replication sharing a common thread pool, continuation allows to send another
+                // batch of updates for the current snapshot sync.
+                if (event.getMetadata().getRequestId() == transitionEventId) {
+                    log.debug("Continuation of log entry sync for {}", event.getEventID());
+                    return this;
+                } else {
+                    log.warn("Unexpected log entry sync continue event {} when in log entry sync state {}.",
+                            event.getEventID(), transitionEventId);
+                }
```

Comment:
this will fall through to default: and then to illegalStateException. Is that intended ?